### PR TITLE
Ideal nav v2 workspaces

### PR DESCRIPTION
**Pull request by WojtekBoman**
### Details

This is the second PR related to the ideal-nav-v2. It introduces:
- Workspace Settings in the FullScreenNavigator
- Account Settings available from the BottomTab
- updated sections on the Settings page (Workspaces List is now always available from AccountSettings)
- updated header and bottom tab

### Fixed Issues

$ https://github.com/Expensify/App/issues/36656
PROPOSAL: N/A

### Tests

- [ ] Verify that no errors appear in the JS console

1. Select any workspace from the workspace switcher
2. Open Account Settings from the bottom tab
3. Verify that sections on the Settings page are updated according to the description in [the issue](https://github.com/Expensify/App/issues/36656)
4. Verify that `w` param is not present in the url in the browser
5. Verify that the profile icon in the bottom tab is selected
6. Open the workspaces list
7. Select any workspace and check if the Workspace Settings screen is displayed in the fullscreen mode (the bottom tab should be hidden)
8. Verify that `w` param is not present in the url in the browser
9. Verify that when you go back to the chat tab, you're still in the workspace selected in the first step (it doesn't work when you open the workspace settings from the deeplink, to test it properly start navigating from the chat tab). 

### Offline tests

N/A

### QA Steps

Same as tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/47774969/c2f15f82-2ddf-4139-ba69-010798a5e3d9

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/47774969/093caa08-205f-4a45-8717-ba8fc20c78fe

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/47774969/83c30ec5-e544-47f5-9ad3-120601421618

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/47774969/8f0bfd16-abf4-4676-b6fc-120308c18a27

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/47774969/24c00fae-81fa-46ba-8000-f1079d654e90

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/47774969/563fa8c4-d8e0-4d7d-abbc-ff0582f7fbb5

</details>

### CODE REVIEW

**Comment by hayata-suenaga** [comment]
thank you for the draft PR @WojtekBoman. please mark the PR ready for review once it's ready 🙇

File: src/pages/home/sidebar/PressableAvatarWithIndicator.js [hunk 1]
```diff
@@ -49,19 +57,23 @@ function PressableAvatarWithIndicator({isCreateMenuOpen, currentUserPersonalDeta
         Navigation.navigate(ROUTES.SETTINGS);
     }, [isCreateMenuOpen]);
 
+    const buttonStyle = isSelected ? {borderWidth: 2, borderRadius: 30, borderColor: theme.splashBG} : {};
```
Comment (kosmydel) [inline src/pages/home/sidebar/PressableAvatarWithIndicator.js:60]: We have to add it to `styles`.

File: src/pages/home/sidebar/PressableAvatarWithIndicator.js [hunk 2]
```diff
@@ -49,19 +57,23 @@ function PressableAvatarWithIndicator({isCreateMenuOpen, currentUserPersonalDeta
...
             onPress={showSettingsPage}
         >
             <OfflineWithFeedback pendingAction={lodashGet(currentUserPersonalDetails, 'pendingFields.avatar', null)}>
-                <AvatarWithIndicator
-                    source={UserUtils.getAvatar(currentUserPersonalDetails.avatar, currentUserPersonalDetails.accountID)}
-                    tooltipText={translate('profilePage.profile')}
-                    fallbackIcon={currentUserPersonalDetails.fallbackIcon}
-                    isLoading={isLoading && !currentUserPersonalDetails.avatar}
-                />
+                <View style={[styles.p1, buttonStyle]}>
```
Comment (kosmydel) [inline src/pages/home/sidebar/PressableAvatarWithIndicator.js:69]:
```suggestion
                <View style={[styles.p1, isSelected && styles.buttonStyle]}>
```

File: src/pages/home/sidebar/AvatarWithOptionalStatus.js [hunk 3]
```diff
@@ -18,14 +18,17 @@ const propTypes = {
 
     /** Emoji status */
     emojiStatus: PropTypes.string,
+
+    isSelected: PropTypes.bool,
```
Comment (kosmydel) [inline src/pages/home/sidebar/AvatarWithOptionalStatus.js:22]: Please add comment

File: src/libs/Navigation/types.ts [hunk 4]
```diff
@@ -478,23 +455,54 @@ type RightModalNavigatorParamList = {
...
-    [SCREENS.SETTINGS.SECURITY]: undefined;
-    [SCREENS.SETTINGS.WALLET.ROOT]: undefined;
-    [SCREENS.SETTINGS.ABOUT]: undefined;
+// type SettingsCentralPaneNavigatorParamList = {
+//     [SCREENS.SETTINGS.PROFILE.ROOT]: undefined;
+//     [SCREENS.SETTINGS.PREFERENCES.ROOT]: undefined;
+//     [SCREENS.SETTINGS.SECURITY]: undefined;
+//     [SCREENS.SETTINGS.WALLET.ROOT]: undefined;
+//     [SCREENS.SETTINGS.ABOUT]: undefined;
+// };
```
Comment (kosmydel) [inline src/libs/Navigation/types.ts:464]:
```suggestion
```

**Comment by shawnborton** [comment]
Let us know when this has some screenshots or a test build to review, thanks! cc @Expensify/design

File: src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/BottomTabBar.tsx [hunk 5]
```diff
@@ -42,12 +42,14 @@ function BottomTabBar({isLoadingApp = false}: PurposeForUsingExpensifyModalProps
...
         const routes = navigationState.routes;
         const currentRoute = routes[navigationState.index];
-        const bottomTabRoute = getTopmostBottomTabRoute(navigationState);
+
         if (
             // When we are redirected to the Settings tab from the OldDot, we don't want to call the Welcome.show() method.
             // To prevent this, the value of the bottomTabRoute?.name is checked here
-            bottomTabRoute?.name === SCREENS.WORKSPACE.INITIAL ||
-            (currentRoute && currentRoute.name !== NAVIGATORS.BOTTOM_TAB_NAVIGATOR && currentRoute.name !== NAVIGATORS.CENTRAL_PANE_NAVIGATOR)
+            // bottomTabRoute?.name === SCREENS.WORKSPACE.INITIAL ||
```
[the remark this answers is not in the document]
Comment (WojtekBoman) [reply src/libs/Navigation/AppNavigator/createCustomBottomTabNavigator/BottomTabBar.tsx:49]: I've asked about it [here](https://github.com/Expensify/App/issues/36656#issuecomment-1971133064)

**Comment by WojtekBoman** [comment]
> Let us know when this has some screenshots or a test build to review, thanks! cc @Expensify/design

I've made it ready for review :) I'm waiting for your comments 👀

**Comment by hayata-suenaga** [comment]
that's a good point

[building right now](https://github.com/Expensify/App/actions/runs/8100282059)

**Comment by WojtekBoman** [comment]
@Expensify/design -> it's ready to test

**Comment by dannymcclain** [comment]
Generally this is feeling pretty sweet! I want to let the other designers review it too but I think it's looking really good so far.

A couple things:
- @Expensify/design do we want the status to display in the avatar tab like it's currently doing in this build? I don't know that we ever talked about/specified that.
- I stumbled onto some weird behavior where if I go to a workspace's settings, goof around a bit, and then click "back", it drops me on the chat tab with the workspace page from the avatar tab in the main view (screenshot+video attached).

CleanShot 2024-03-01 at 08 07 09@2x

https://github.com/Expensify/App/assets/7516624/8ab80ed5-e149-4ee0-a2b8-d3c11e2a2fb2

Basically all I do to reproduce this is navigate to avatar tab > click on workspaces > go into a workspace's settings > click on description (or anything really), close the RHP > click the top left back arrow.

**Comment by WojtekBoman** [comment]
@dannymcclain Thanks for noticing [that](https://github.com/Expensify/App/pull/37421#issuecomment-1973272942)! I've added some fixes and I now it should works. You can check it when the new build is ready :)

**Comment by hayata-suenaga** [comment]
thank you Danny for reporting the issue 😄

**Comment by shawnborton** [comment]
Gonna fire up a new test build and then I'll also take it for a spin! But so far this is looking great!

**Comment by WojtekBoman** [comment]
@mkhutornyi Could you take a look at it when you have time? :)

**Comment by shawnborton** [comment]
Are we able to use a higher quality image for the avatar photo? It's pretty blurry:
CleanShot 2024-03-04 at 11 34 42@2x

**Comment by shawnborton** [comment]
I think we should add 12px padding below the left side's header here, so that the first active item in the list lines up perfectly with the top of the card in the right content view:
CleanShot 2024-03-04 at 11 39 54@2x

**Comment by shawnborton** [comment]
Not sure if this was changed here, but I think we want the "Switch to Expensify Classic" to be at the top of Settings and not use an arrow icon. I think we're handling this elsewhere, just wanted to call it out though. We might just need to pull main.
CleanShot 2024-03-04 at 11 41 51@2x

**Comment by shawnborton** [comment]
> @Expensify/design do we want the status to display in the avatar tab like it's currently doing in this build? I don't know that we ever talked about/specified that.

@dannymcclain I think I kinda like it there? I don't feel too strongly though... I can see where it makes things a little busy. What do you think?

**Comment by shawnborton** [comment]
For the bottom tabs, I think in Figma we have them at 72px tall, but they are 80px in this PR:
CleanShot 2024-03-04 at 11 43 50@2x

Should we use this as an opportunity to slim them down slightly? We just need to shave 4px off of the top padding and bottom padding. We'd still have the safeSpace below them on the mobile devices.

**Comment by shawnborton** [comment]
Okay I think that's really all I got - looking good to me so far!

**Comment by dannymcclain** [comment]
>I think we should add 12px padding below the left side's header here, so that the first active item in the list lines up perfectly with the top of the card in the right content view:

Stupid detail: let's add that 12px to the top of the nav items list. That's the way we do it in the center pane (12px top padding on the "content" container)

>do we want the status to display in the avatar tab like it's currently doing in this build? I don't know that we ever talked about/specified that.
>
>@dannymcclain I think I kinda like it there? I don't feel too strongly though... I can see where it makes things a little busy. What do you think?

I kinda do too haha. On desktop it makes it SUPER fast to get to your status which I loooooove. On mobile it will likely never be practical to actually tap it, but that's probably ok. But you're right, it is fairly busy... So I can see us going either way.

>Should we use this as an opportunity to slim them down slightly? We just need to shave 4px off of the top padding and bottom padding. We'd still have the safeSpace below them on the mobile devices.

I definitely think we should. Our bottom tabs are pretty big compared to basically all other apps on my phone. I assume they are all using native iOS tabs—so I'm not saying they need to match, just calling out how much bigger ours are (and in this comparison I have ours at 72px).
image

**Comment by shawnborton** [comment]
> On desktop it makes it SUPER fast to get to your status which I loooooove. On mobile it will likely never be practical to actually tap it, but that's probably ok. But you're right, it is fairly busy... So I can see us going either way.

Regardless of what we decide, I think I would make it so that tapping this part of the bottom tab just always takes you to Settings. I don't think we should ever block on the user getting taken to the parent level of Settings on based on exactly where they tap on this tab. So I'd be cool to keep it, but always navigate to Settings. From here you will see the status in the top right for easy access. 

Curious what Judge Jon thinks though of course!

**Comment by shawnborton** [comment]
And nice, let's get these tabs down to 72px then and see how it feels from there. Thanks for that comparison btw!

**Comment by dannymcclain** [comment]
>Regardless of what we decide, I think I would make it so that tapping this part of the bottom tab just always takes you to Settings. I don't think we should ever block on the user getting taken to the parent level of Settings on based on exactly where they tap on this tab. So I'd be cool to keep it, but always navigate to Settings. From here you will see the status in the top right for easy access.
>
>Curious what Judge Jon thinks though of course!

Now that you say that, I'm down with this approach and agree with you. If we keep it, it should basically just be a little visual indicator instead of an interactive element. Knowing Jon, I have a feeling he will agree with you 😁 but I can't wait to find out!

**Comment by WojtekBoman** [comment]
> I think we should add 12px padding below the left side's header here, so that the first active item in the list lines up perfectly with the top of the card in the right content view: CleanShot 2024-03-04 at 11 39 54@2x

I've checked it, and to set it at the same line, padding should be 8px. This is how it looks after the refactor: 
image
Is it ok for you?

**Comment by shawnborton** [comment]
Ah sorry, I don't mean that we should change the padding of the nav item itself. But rather, the space between the left header area and the first item needed a gap of 12px. The right side should also have the 12px gap.

**Comment by shawnborton** [comment]
So both the left header and the right header should be 80px tall:
CleanShot 2024-03-05 at 09 09 27@2x

With 12px padding below:
CleanShot 2024-03-05 at 09 10 06@2x

CleanShot 2024-03-05 at 09 10 20@2x

**Comment by shawnborton** [comment]
Looks like the border color on the outline of the pencil button is the wrong color:
CleanShot 2024-03-05 at 15 15 47@2x

**Comment by shawnborton** [comment]
It looks like the wordmark is not aligned with the "Chats" label, likely because of the adhoc label?
CleanShot 2024-03-06 at 08 26 02@2x

**Comment by WojtekBoman** [comment]
> It looks like the wordmark is not aligned with the "Chats" label, likely because of the adhoc label? CleanShot 2024-03-06 at 08 26 02@2x

Simulator Screenshot - iPhone 15 Plus - 2024-03-06 at 15 43 23

I think it's a problem that occurs also on the main branch:

**Comment by mountiny** [comment]
@mkhutornyi ready for testing, creating a new build https://github.com/Expensify/App/actions/runs/8189318520

**Comment by shawnborton** [comment]
This is feeling pretty solid to me! cc @Expensify/design in case ya'll wanna giver a testy run too.

**Comment by shawnborton** [comment]
I mentioned this before but it still feels like the big avatar here is low resolution:
CleanShot 2024-03-07 at 11 17 49@2x

Can we fix that here or do we need a follow up?

**Comment by dannymcclain** [comment]
Feeling good to me too! 🚀 

One thing I wanted to call out that I know Shawn has mentioned before: currently the status icon in the bottom tab is a separate interactive element from the tab itself—do we want that? I know Shawn has mentioned that the we should keep that whole thing as a single interactive tab and not have a direct link to the status in the tab bar. Is that something we should deal with in this PR or as a follow up?

CleanShot 2024-03-07 at 10 24 34

**Comment by WojtekBoman** [comment]
> I mentioned this before but it still feels like the big avatar here is low resolution: CleanShot 2024-03-07 at 11 17 49@2x
> 
> Can we fix that here or do we need a follow up?

This issue is also present on the main, so maybe we can create the follow up

**Comment by shawnborton** [comment]
Ah good shout - we should solve that here, and I agree we should just make that entire tab always navigate you to settings

**Comment by dannymcclain** [comment]
>Profile tab tooltip position is inconsistent with others. Expected?

I think they should all be the same. @Expensify/design?

**Comment by WojtekBoman** [comment]
@dannymcclain @shawnborton 

https://github.com/Expensify/App/assets/47774969/76619eb6-facb-4431-8ee4-702f56eb4e20

Will it be ok? I can also remove a tooltip from the status icon, what do you think?

**Comment by shawnborton** [comment]
I agree, we should remove the tooltip in that case

**Comment by dannymcclain** [comment]
>Will it be ok? I can also remove a tooltip from the status icon, what do you think?

@WojtekBoman I think we need to remove the tooltip from status and make sure all the tab tooltips are positioned the same.

**Comment by WojtekBoman** [comment]
> > Will it be ok? I can also remove a tooltip from the status icon, what do you think?
> 
> @WojtekBoman I think we need to remove the tooltip from status and make sure all the tab tooltips are positioned the same.

I've already added the fix :) Now it works this way: 

https://github.com/Expensify/App/assets/47774969/34f76d12-b0f2-41b1-bebc-f238be934da3

**Comment by dannymcclain** [comment]
Fantastic—looks great!

Quick question for @Expensify/design: does it make more sense for the tooltips to be anchored to the icon (as shown in [this video](https://github.com/Expensify/App/pull/37421#issuecomment-1985847120)) or anchored to the top of the _tab_?

**Comment by mountiny** [comment]
Making a new build https://github.com/Expensify/App/actions/runs/8207749469

File: src/components/AvatarSkeleton.tsx [hunk 6]
```diff
@@ -1,21 +1,25 @@
 import React from 'react';
 import {Circle} from 'react-native-svg';
 import useTheme from '@hooks/useTheme';
+import variables from '@styles/variables';
 import SkeletonViewContentLoader from './SkeletonViewContentLoader';
 
 function AvatarSkeleton() {
     const theme = useTheme();
+
```
Comment (mountiny) [inline src/components/AvatarSkeleton.tsx:9]:
```suggestion
```

File: src/libs/Navigation/AppNavigator/createCustomFullScreenNavigator/CustomFullScreenRouter.tsx [hunk 7]
```diff
@@ -6,41 +6,49 @@ import type {FullScreenNavigatorRouterOptions} from './types';
...
-            state.routes.unshift({name: SCREENS.SETTINGS.ROOT});
+            state.routes.unshift({
+                name: SCREENS.WORKSPACE.INITIAL,
+                params: topmostWorkspaceCentralPaneRoute?.params ?? workspacesCentralPaneParams,
+            });
         }
     }
 
     // If the screen is wide, there should be at least two screens inside:
     // - SETINGS.ROOT to cover left pane.
```
Comment (mountiny) [inline src/libs/Navigation/AppNavigator/createCustomFullScreenNavigator/CustomFullScreenRouter.tsx:36]: Right? Or how should this comment be updated?
```suggestion
    // - WORKSPACE.INITIAL to cover left pane.
```

File: src/pages/home/sidebar/AvatarWithOptionalStatus.js [hunk 8]
```diff
@@ -18,44 +15,43 @@ const propTypes = {
 
     /** Emoji status */
     emojiStatus: PropTypes.string,
+
+    /** Whether the avatar is selected */
+    isSelected: PropTypes.bool,
+
+    onPress: PropTypes.func,
```
Comment (mountiny) [inline src/pages/home/sidebar/AvatarWithOptionalStatus.js:22]: Please add a comment here just to follow the style guide

File: src/pages/home/sidebar/BottomTabAvatar.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+import Navigation from '@libs/Navigation/Navigation';
+import ROUTES from '@src/ROUTES';
+import AvatarWithOptionalStatus from './AvatarWithOptionalStatus';
+import PressableAvatarWithIndicator from './PressableAvatarWithIndicator';
+
+const propTypes = {
+    /** Whether the create menu is open or not */
+    isCreateMenuOpen: PropTypes.bool,
+
+    isSelected: PropTypes.bool,
```
Comment (mountiny) [inline src/pages/home/sidebar/BottomTabAvatar.js:15]: Pleaase add a comment

File: src/pages/home/sidebar/BottomTabAvatar.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable rulesdir/onyx-props-must-have-default */
```
Comment (mountiny) [inline src/pages/home/sidebar/BottomTabAvatar.js:1]: This is a new file, can you write it in TS?

File: src/pages/home/sidebar/PressableAvatarWithIndicator.js [hunk 11]
```diff
@@ -1,67 +1,64 @@
...
     /** The personal details of the person who is logged in */
     currentUserPersonalDetails: personalDetailsPropType,
 
     /** Indicates whether the app is loading initial data */
     isLoading: PropTypes.bool,
+
+    /** Whether the avatar is selected */
+    isSelected: PropTypes.bool,
+
+    onPress: PropTypes.func,
```
Comment (mountiny) [inline src/pages/home/sidebar/PressableAvatarWithIndicator.js:29]: Please add a comment

**Review by mountiny — CHANGES_REQUESTED** [review]
Great job so far

**Review by mkhutornyi — COMMENTED** [review]
Bug1:  No left pane on workspace profile page


https://github.com/Expensify/App/assets/97676131/c34ee239-97a1-4abc-a83f-89316f1a10ba

----

Bug2: navigate to wrong chat (Concierge) after creating new room


https://github.com/Expensify/App/assets/97676131/18b8a53e-8dcf-4171-8b4a-244b165448e0

----

Bug3: anonymous user is able to navigate to Profile tab

bug3

----

Bug4: breadcrumb **root** is always "Expensify" even after selecting workspace


https://github.com/Expensify/App/assets/97676131/99ada3ed-e289-46ac-ac09-1275d6f23450

----

Bug5: [mobile] wrong navigation animation when go back from workspace profile


https://github.com/Expensify/App/assets/97676131/8188afce-23bd-4325-831f-1e916f3b4eef

**Comment by WojtekBoman** [comment]
> Bug1: No left pane on workspace profile page
> 
>  bug1.mov 

Try reinstalling node_modules. It should help, because in this PR has been introduced changes in the patch file responsible for displaying left pane in FullScreenNavigator and you need reinstall packages to apply these changes

**Comment by mountiny** [comment]
@mkhutornyi the adhoc builds should be accessible now

**Comment by WojtekBoman** [comment]
> Bug2: navigate to wrong chat (Concierge) after creating new room

https://github.com/Expensify/App/assets/47774969/c33ba3ac-1d94-4cb3-a518-a624acffb437

I can't reproduce this one. Are there any other steps I need to take to reproduce this?

**Comment by mountiny** [comment]
@mkhutornyi thanks for initial reports. hopefully you can complete the first round today 🙇

**Comment by mountiny** [comment]
@mkhutornyi can you repro on staging too?

File: src/components/Indicator.tsx [hunk 12]
```diff
@@ -30,16 +40,25 @@ type IndicatorOnyxProps = {
...
 
-function Indicator({bankAccountList, fundList, userWallet, walletTerms, loginList}: IndicatorOnyxProps) {
+function Indicator({reimbursementAccount, allPolicyMembers, policies, bankAccountList, fundList, userWallet, walletTerms, loginList}: IndicatorOnyxProps) {
     const theme = useTheme();
     const styles = useThemeStyles();
 
+    // If a policy was just deleted from Onyx, then Onyx will pass a null value to the props, and
+    // those should be cleaned out before doing any error checking
+    const cleanPolicies = Object.fromEntries(Object.entries(policies ?? {}).filter(([, policy]) => policy?.id));
+    const cleanAllPolicyMembers = Object.fromEntries(Object.entries(allPolicyMembers ?? {}).filter(([, policyMembers]) => !!policyMembers));
```
[the remark this answers is not in the document]
Comment (WojtekBoman) [reply src/components/Indicator.tsx:50]: It's logic copied from the `hasGlobalWorkspaceSettingsRBR` method of `WorkspacesSettingsUtils.ts`. I added this in another PR. In policies we also sometimes store (when trying to open a non-existent policy) invalid policies that are not displayed on the list, so we don't want to show an indicator for them

**Comment by WojtekBoman** [comment]
@mountiny @mkhutornyi I've solved bugs reported in the comments that aren't present on the main branch :)

**Comment by WojtekBoman** [comment]
@mkhutornyi Conflicts have been solved. Please ping me, if they appear again :)

**Comment by WojtekBoman** [comment]
@mkhutornyi Conflicts have been resolved :) Ping me, if they appear again :)

**Comment by trjExpensify** [comment]
> Anonymous user is still able to access profile page via deep link. This also happens on staging. Is this bug or expected? If bug, can we create new GH for this?
> 
>  Screen.Recording.2024-03-12.at.1.06.47.pm.mov

@mkhutornyi that isn't expected, but unrelated to this PR, right? Looks like @DylanDylann helped fixed it once [here](https://github.com/Expensify/App/issues/28925). My suspicion is that the TS migration might have caused the regression. Couple of questions: 

- Is it staging and prod, or staging only? 
- What are the steps you're taking? Are they: 

1. Sign-out of any instance of Expensify you have on the device
2. Click a public room URL as an anonymous user like: https://staging.new.expensify.com/r/868417741568295
3. Enter https://staging.new.expensify.com/settings/profile in the browser
4. Observe you land on the profile page 

- Is it just the profile page once you're on the profile page, or can you access other pages in settings without being prompted to sign-in (or via deep link to the others)?

**Comment by WojtekBoman** [comment]
image
This issue with wrong display of the avatar in the bottom tab doesn't occur for me, I've increased borderRadius of the view that is a wrapper for the avatar, so maybe it will help. [This one](https://github.com/Expensify/App/pull/37421#issuecomment-1991524309) should work now.

**Comment by WojtekBoman** [comment]
IMG_7372
 On a physcial device it also works for me, I don't know what can be a reason of this issue

**Comment by WojtekBoman** [comment]
@mkhutornyi Conflicts solved :)

**Comment by mountiny** [comment]
New build is running

File: src/pages/workspace/WorkspaceInitialPage.tsx [hunk 13]
```diff
@@ -216,39 +217,46 @@ function WorkspaceInitialPage({policyDraft, policy: policyProp, policyMembers, r
...
             testID={WorkspaceInitialPage.displayName}
-            includePaddingTop={false}
             includeSafeAreaPaddingBottom={false}
             style={[styles.pb0]}
         >
             <FullPageNotFoundView
-                onBackButtonPress={PolicyUtils.goBackFromInvalidPolicy}
-                onLinkPress={PolicyUtils.goBackFromInvalidPolicy}
+                onBackButtonPress={Navigation.dismissModal}
+                onLinkPress={Navigation.resetToHome}
```
[the remark this answers is not in the document]
Comment (WojtekBoman) [reply src/pages/workspace/WorkspaceInitialPage.tsx:241]: Ok, I can change it, but in some cases, for example when you open a link in your browser to the non-existing workspace, it won't navigate you to the home page but will open the list of workspaces. Is it ok?

[the remark this answers is not in the document]
Comment (WojtekBoman) [reply src/pages/workspace/WorkspaceInitialPage.tsx:241]: I think it can be a problem with builds on staging, is it occurs also on your local build?

[the remark this answers is not in the document]
Comment (WojtekBoman) [reply src/pages/workspace/WorkspaceInitialPage.tsx:241]: @mountiny As far as I know, we had a problem with builds that not-found page appears when we open the app, am I right?

[the remark this answers is not in the document]
Comment (WojtekBoman) [reply src/pages/workspace/WorkspaceInitialPage.tsx:241]: I believe this problem with builds causes this issue. It seems to work fine locally

https://github.com/Expensify/App/assets/47774969/2b3cfb38-21f8-45b8-96a2-79d57ae74ca5

**Review by mkhutornyi — APPROVED** [review]
Given that no blocker issues, I am going to approve.
Since this is big refactor, we might have missed edge cases which should be handled as follow-up.

File: src/pages/workspace/WorkspaceInitialPage.tsx (hunk 13 above)
[the remark this answers is not in the document]
Comment (mountiny) [reply src/pages/workspace/WorkspaceInitialPage.tsx:241]: Yep this seems like issue with adhoc builds

**Review by mountiny — APPROVED** [review]
Changes look good to me too @Expensify/design I believe this one has been checked by design already, but anything else to check for before we merge?

**Review by mountiny — CHANGES_REQUESTED** [review]
image

in desktop view I am not seeing the workspace settings options

**Comment by shawnborton** [comment]
Not sure how we keep getting this regression, but the border color on the edit workspace profile pic button is incorrect:
CleanShot 2024-03-12 at 17 16 30@2x


Could have sworn that @narefyev91 fixed that - maybe we just need to pull main?

**Comment by mountiny** [comment]
@mkhutornyi I see thanks!

**Comment by mountiny** [comment]
Seems to e fixed for me 
image

**Review by mountiny — APPROVED** [review]
Ok this works well for me now, I will go ahead and merge this before there is more conflicts

**Comment by shawnborton** [comment]
@mountiny in your screenshot, the border color is still borked:
CleanShot 2024-03-13 at 08 52 00@2x


Can we please follow up and fix this?

**Comment by shawnborton** [comment]
Anyone else seeing this small gap on the top of the Settings view?

CleanShot 2024-03-13 at 17 22 47@2x

**Comment by kosmydel** [comment]
> @kosmydel should be a quick one for you to follow up on tomorrow

I've prepared [a PR](https://github.com/Expensify/App/pull/38274) with a fix.

**Comment by godofoutcasts94** [comment]
This PR has caused this [issue](https://github.com/Expensify/App/issues/38436#issuecomment-2002332908)

**Comment by akinwale** [comment]
This PR caused a [regression](https://github.com/Expensify/App/issues/38296#issuecomment-1997605199).

**Comment by jjcoffee** [comment]
This caused a regression in https://github.com/Expensify/App/issues/38277. When deleting a policy, we needed to add a check if the policy selected in the switcher is the one being deleted, so that we can reset the switcher to `All workspaces`. More details [here](https://github.com/Expensify/App/issues/38277#issuecomment-1998522263).

File: src/libs/Navigation/linkingConfig/getAdaptedStateFromPath.ts [hunk 14]
```diff
@@ -165,18 +171,25 @@ function getAdaptedState(state: PartialState<NavigationState<RootStackParamList>
 
         if (topmostNestedRHPRoute) {
             let matchingRootRoute = getMatchingRootRouteForRHPRoute(topmostNestedRHPRoute);
-
+            const isRHPScreenOpenedFromLHN = topmostNestedRHPRoute?.name && RHP_SCREENS_OPENED_FROM_LHN.includes(topmostNestedRHPRoute?.name as RHPScreenOpenedFromLHN);
```
Comment (c3024) [inline src/libs/Navigation/linkingConfig/getAdaptedStateFromPath.ts:174]: `Track expense` adds self DM as matching route and it should be added in the RHPScreensOpenedFromLHN. Fixed with #46848 . Issue here #46674.

File: src/pages/workspace/WorkspaceInitialPage.tsx (hunk 13 above)
[the remark this answers is not in the document]
Comment (eh2077) [reply src/pages/workspace/WorkspaceInitialPage.tsx:241]: > Ok, I can change it, but in some cases, for example when you open a link in your browser to the non-existing workspace, it won't navigate you to the home page but will open the list of workspaces. Is it ok?

We should navigate back the list of workspace, which is reported here https://github.com/Expensify/App/issues/61485
